### PR TITLE
Adds test coverage to new bakery

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7197/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7198/

File: jujugui/static/gui/src/test/test-bakery.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+      },
+      removeItem: function(item) {
+        delete this.store[item];
+      }
+    };
+
+    client = {
+      _sendRequest: sinon.stub(),
+      sendGetRequest: function(...args) {
+        this._sendRequest('get', ...args);
```
Comment (frankban) [inline jujugui/static/gui/src/test/test-bakery.js]: Nice!

File: jujugui/static/gui/src/test/test-bakery.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+      fakeLocalStorage, {services: {charmstore: 'http://example.com/'}});
+    bakery = new Y.juju.environments.web.Bakery(client, storage);
+  });
+
+  afterEach(() => {
+    bakery = null;
+  });
+
+  it('exists', () => {
+    bakery = new Y.juju.environments.web.Bakery({});
```
Comment (frankban) [inline jujugui/static/gui/src/test/test-bakery.js]: This line isn't really required, we already have a bakery at this point, correct?

File: jujugui/static/gui/src/test/test-bakery.js [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    assert.isObject(bakery);
+  });
+
+  describe('can send requests', () => {
+    it('sets the method', () => {
+      bakery.sendRequest('http://example.com/', 'GET');
```
Comment (frankban) [inline jujugui/static/gui/src/test/test-bakery.js]: I'd use an array for testing this, and also add more arguments? Something like
```javascript
const url = 'http://example.com/';
const headers = {header: 42};
const callback = response => 42;
let body = 'content';
let args;
['PATCH', 'post', 'PUT'].forEach(method => {
  bakery.sendRequest(url, method, header, body, callback);
  assert.equal(client._sendRequest.callCount, 1);
  args = client._sendRequest.args[0];
  assert.equal(args[0], method.toLower());
  assert.equal(args[1], url);
  assert.equal(args[2], {header: 42, 'Bakery-Protocol-Version': 1});
  assert.equal(args[3], body);
  assert.equal(args[4], bakery._wrapCallback(callback));
  client._sendRequest.reset();
}):
['GET', 'delete'].forEach(method => {
  bakery.sendRequest('http://example.com/', method, callback);
  assert.equal(client._sendRequest.callCount, 1);
  args = client._sendRequest.args[0];
  assert.equal(args[0], method.toLower());
  assert.equal(args[1], url);
  assert.equal(args[2], {header: 42, 'Bakery-Protocol-Version': 1});
  assert.equal(args[3], bakery._wrapCallback(callback));
  client._sendRequest.reset();
}):
```
What do you think?

File: jujugui/static/gui/src/test/test-bakery.js [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+        assert.equal(msg, `discharge failed: ${error.message}`);
+      };
+      bakery.discharge('macaroons', success, failure);
+    });
+
+    it('handles third party discharge', () => {
+      const condition = 'this is a caveat';
+      const success = macaroons => {}; 
+      const failure = err => {};
+      bakery._getThirdPartyDischarge(
```
Comment (frankban) [inline jujugui/static/gui/src/test/test-bakery.js]: Is it possible to test this only using the public API? So basically passing a macaroon with a third party?
Anyway, thanks for these tests, it feels like we are already exercising much more than before.

Comment (jaycee) [inline jujugui/static/gui/src/test/test-bakery.js]: ...maybe. Generating a valid macaroon for this test was not something I wanted to get into, but it's probably worth looking into.

File: jujugui/static/gui/src/test/test-bakery.js (hunk 3 above)
Comment (jaycee) [inline jujugui/static/gui/src/test/test-bakery.js]: I like it, I'll make that update.

File: jujugui/static/gui/src/test/test-bakery.js (hunk 2 above)
Comment (jaycee) [inline jujugui/static/gui/src/test/test-bakery.js]: Yeah, this didn't get updated after I moved bakery creation into beforeEach

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7213/

**Pull request merged**